### ISSUE TITLE
Some bindings don't work (set-speed axis, rot damping on/off)

### ISSUE THREAD

**Issue by Keeper1st** [pioneerspacesim/pioneer#4532]
### Observed behaviour
BindAxisThrottle in the Speed Mode section of the control options does nothing. (Strangely, as well as an axis binding, it has two button bindings, which also do nothing.) The only way to adjust the Set Speed Mode value is through the buttons set as BindIncreaseSpeed and BindDecreaseSpeed.

In addition, BindToggleRotationDamping in the Miscellaneous section also does not actually do anything. One must use the on-screen button to toggle rotation damping.

Note that in both cases, the options system does recognize an axis or button, and allows them to be set. They just don't actually work within the game.

### Expected behaviour
The axis set on BindAxisThrottle should adjust the Set Speed value. There seems no need for the two button bindings also settable within BindAxisThrottle.

The button assigned to BindToggleRotationDamping should toggle the rotation damping on/off setting.

### Steps to reproduce
Assign buttons/axes and try it!

__My pioneer version (and OS):__
Built from master on 7 February 2019, compiled for Windows x64.

**Comment by nozmajner** [comment]
I can confirm. 
The Rotation damping does not toggle the setting. 
The Set speed key bindings work for me, (T and G by default) but the Negative and Positive buttons in the Set Speed axis do not. (What are these positive and negative are anyway?)
I can't test the axis itself right now.

**Comment by Keeper1st** [comment]
Yes, BindIncreaseSpeed and BindDecreaseSpeed buttons do work. It's BindAxisThrottle that does not (and as you say, it has mysterious Positive and Negative button values as well which do nothing and probably should not be there as this is an axis setting).

**Comment by Web-eWorks** [comment]
BindAxisThrottle is not actually implemented at the moment. It is there as a future binding for using a joystick throttle with the set-speed control mode. It will eventually replace the dedicated increase / decrease speed buttons. The Positive and Negative bindings are endemic to all axis bindings, and allow binary control over the axis.

ToggleRotationDamping is likely non-functional from before the input binding revamp, but I'll look into it soon.

**Comment by Keeper1st** [comment]
I don't think it should replace the set-speed increase/decrease buttons -- just be available if one wishes to use an axis instead. For example, currently I have my throttle mapped to the forward and reverse thrusters (I'd prefer it to be the forward thruster only as that's the hugely power thruster that makes small maneuvers difficult -- which is why I think a "landing mode" when the landing gear is down should be implemented that limits the forward thruster output to whatever the reverse thruster output is on the ship). Using my throttle for the main thruster means I do have to use the buttons when adjusting set-speed mode.

Of course the other thing I'd like to have is the internal camera control to be mappable to X/Y axes, just like the external camera is (only of course it would be translated to absolute direction up to 180 degrees each way, whereas the external camera axis acts as a movement speed adjuster).

**Comment by Web-eWorks** [comment]
@Keeper1st an 'axis' under the new input system is a combination of a joystick axis input and two key inputs for 1 and -1 respectively. Merging the arbitrary buttons with an axis will not actually lose the ability to bind buttons to increasing / decreasing the set-speed value, but simply adds a joystick axis to the buttons and makes the code dedicated to reading that input simpler.

Limiting the main thrusters to the common-thrust-value of the ship is planned, but is slated for after a major rework of the game's internal code.

Regarding internal camera control, do you mean that the axis would switch to the next camera position in that direction (i.e. press KP-left once to get the left camera, press again to get the rear camera?) or are you asking for something else? I can't quite parse what you mean.

**Comment by Keeper1st** [comment]
No, I mean the axis would be an alternative to mouselook -- just with obvious maximum limits in each direction. Just as you can use an axis to look around the ship with the external camera, if one could use an axis to look around in the internal camera that would suit two types of users: Those with a gamepad who could use one stick as a head-mover (or those who fly with the mouse but do have a joystick handy), and those with a head-tracker who easily could set up their system as a virtual joystick. (I'm in the latter camp!) I figure ideally an axis' data would translate such that a full stick to the right, for example, would equate to 180 degrees to the right from the selected camera's center, so you'd look behind you from the front camera, or in front of you from the rear camera, for example. And of course any value in between.

**Comment by Web-eWorks** [comment]
I definitely want to get that in. It's now on [The List](https://pioneerwiki.com/wiki/User:Sturnclaw), though it may be a while before I get around to it.

**Comment by Keeper1st** [comment]
Cool! In the mean time, I've edited the mouselook in my source to remove the 45x multiplier, have my head tracker set up as a virtual mouse, with a toggle button that holds the middle mouse button down. I just have to turn toggle it off whenever I need actually to use the mouse. (Modified the FOV in my Config file to a realistic setting as well since now I can look around. Game looks so much cooler when you can see planets from their moons etc.! Had to make my own high-res skybox for it to eliminate big white splotches.)

**Comment by Web-eWorks** [comment]
@Keeper1st turns out "a while" is 22 days. See #4536 for the headtracking implementation.

**Comment by impaktor** [comment]
@Web-eWorks can we close this? I.e. your patch also fixes this issue? (if problem popps up again, keepr can re-open)

**Comment by Web-eWorks** [comment]
@impaktor nope, sorry. Only implemented headtracking, though I'll see about resolving this issue soon.
